src/ut_main.c

```diff
@@ -72,18 +78,30 @@ static void usage( void )
 static bool decodeOptions( int argc, char **argv )
 {
     int opt;
+    int option_index = 0;
     ut_kvp_status_t status;
 
     memset(&gOptions,0,sizeof(gOptions));
 
     /* Console mode is always enabled we don't need a switch for that */
     gOptions.testMode = UT_MODE_CONSOLE;
-
+    
+    // Define long options
+    static struct option long_options[] = {
+        {"gtest_output", required_argument, 0, 0}, // 0 is used as a placeholder for gtest_output
+        {0, 0, 0, 0} // Terminator
+    };
+
+#ifdef UT_CUNIT
     /* Set the filepath always by default to /tmp/ */
     UT_log_setLogFilePath(UT_LOG_DEFAULT_PATH); /* Use Macro */
     UT_set_results_output_filename( UT_log_getLogFilename() );
+#else
+    /* Set the filepath always by default to /tmp/ */
+    UT_log_setLogFilePath((char* )UT_LOG_DEFAULT_PATH);
```

Comment:
UT_log_setLogFilePath either takes a char* on both or takes a #define on both.

please fix the function to correct it.